src/fast_reader_bseq.c

```
@@ -113,7 +118,22 @@ bseq1_t *bseq_read_fast(int64_t chunk_size, int *n_, void *ks1_, void *ks2_, int
             break;
         }
         if (n >= m) {
-            m = m ? m << 1 : 256;
+            if (m == 0) {
+                /* L8: size the buffer once from the chunk's base budget and the
+                 * first read length, instead of doubling 256->512->... up to the
+                 * final count (~8-10 reallocs + full copies per chunk on the
+                 * serial read thread). `size` counts bases across all reads and
```

Cppcheck (2.21.0)

[style] The function 'kputl' is never used.

(unusedFunction)